src/linux.rs

```diff
@@ -60,24 +60,59 @@ impl Output {
 
 #[cfg(target_os = "linux")]
 pub(crate) fn get_disk_id() -> Result<String, HWIDError> {
-    let mut com = Command::new("sh");
-    com.arg("-c").arg("lsblk -f -J -o NAME,MOUNTPOINT,UUID");
+    let output = run_command("lsblk -f -J -o NAME,MOUNTPOINT,UUID")?;
 
-    let output = com.output()?;
-
-    let output_string = String::from_utf8(output.stdout)?;
+    let output_string = String::from_utf8(output.into())?;
     let parsed: Output = serde_json::from_str(output_string.as_str())?;
     let uuid = parsed.get_root()?;
     Ok(uuid)
 }
 
+#[cfg(target_os = "linux")]
+fn run_command(command: &str) -> Result<String, HWIDError> {
+    let mut cmd = Command::new("sh");
+    let cmd = cmd.arg("-c")
+        .arg(command);
+
+    let output = cmd.output()?;
+    if !cmd.status()?.success() {
+        return Err(HWIDError::new(
+            &format!("Failed to run command: {command}"),
+            &String::from_utf8(output.stderr.into())?,
+        ));
+    }
+
+    Ok(String::from_utf8(cmd.output()?.stdout)?)
+}
+
+#[cfg(target_os = "linux")]
+fn get_mac_addressof_interface(interface_name: &str) -> Result<String, HWIDError> {
+    run_command(&format!("cat /sys/class/net/{interface_name}/address"))
+}
```

Comment:
Great work here, the original code was expecting all commands to always succeed, which was hiding the error mentioned in #16.